### ISSUE TITLE
Donations

### ISSUE THREAD

**Issue by wsw70** [dani-garcia/bitwarden_rs#707]
This is fantastic work, thank you very much.

I took the premium account with Bitwarden (to support them and their reasonable approach) even though I do not need it with your implementation  - now it is bitwarden_rs' turn: **how to make a donation?**

**Comment by mprasil** [comment]
Last time we had this discussion on our matrix chat, the consensus was that it's upstream that really needs money as we rely on it quite a lot. The result was the [support wiki page](https://github.com/dani-garcia/bitwarden_rs/wiki/Supporting-upstream). As for supporting this project, it seems that time is the rare resource we all lack. So helping others resolve problems with the issues (many questions aren't that technical), fixing bugs (if you feel like writing some rust code) and such.

Not sure if anything changed since then.

**Comment by mprasil** [comment]
@dani-garcia anything to add here?

**Comment by dani-garcia** [comment]
Well so far I always guided people to support upstream bitwarden, but never thought about someone who might want to support both upstream bitwarden AND bitwarden_rs.

As @mprasil said helping with issues or with the wiki is always appreciated, of course.

Personally I'm not in need of monetary donations at the moment, thankfully. That might change if I ever need to implement another two factor hardware key, though so far community testing has helped in this spot. It could also change if for whatever reason we had to fork the mobile clients, where I'd need a license to publish them, but I already have limited time so let's hope that doesn't happen.

The other possible use for donations would be for a third party audit or for bug bounties, but those aren't cheap, so I don't know how realistic these options would be for us. I mean we could offer small sums to big pull requests like the DB backend implementations or the LDAP support, or for helping with complicated issues like BlackDex is doing with the android login issue, but I'm not sure the money we'll get is enough to sway someone to contribute to the project.

Also now that you bring it up, Github Sponsors was released recently and they have no fees, so maybe we can consider enabling it, and depending on what we get we can figure what to do? We could definitely use more build workers on docker hub at least, haha.

**Comment by mprasil** [comment]
Docker hub is great idea. $7/month would give us 5 parallel builds. So we could have all images built in about 2h instead of waiting half day.

**Comment by dani-garcia** [comment]
So I followed on the GitHub sponsors idea after we had this convo, and it's been approved now, available here:
https://github.com/sponsors/dani-garcia

A bit dissapointingly, they only allow subscriptions and not one time donations, but that can be worked around by the users by just unsubscribing afterwards.

The way Sponsors seems to work is with rigid tiers which seems limited, but I added a couple of low cost options which are the ones I expect most users to take, then a couple of higher cost ones which might allow us to apply that bug bounty idea I had, though I don't expect most users if any to go that high.

I'd appreciate any comments and suggestions in how to improve this.

As an aside, for the first year Github will match any user contribution, that's really nice of them.
